Docs: Add theme dependency and instructions (#4747)
p4lang/p4c: docs/README.md

Documenting the workings of the compiler is a never-ending (many times
overlooked) job. We can always write better documentation!

> In P4C, documentation is generated using Doxygen.The generated documentation depends on [Doxygen Awesome CSS](https://github.com/jothepro/doxygen-awesome-css). The documentation is dynamically updated and deployed on [GitHub Pages](https://p4lang.github.io/p4c/).

Reviewer: ```suggestion
In P4C, documentation is generated using Doxygen. The generated documentation depends on [Doxygen Awesome CSS](https://github.com/jothepro/doxygen-awesome-css). The documentation is dynamically updated and deployed on [GitHub Pages](https://p4lang.github.io/p4c/).
```

> In P4C, documentation is generated using Doxygen.he generated documentation is enhanced with [Doxygen Awesome CSS](https://github.com/jothepro/doxygen-awesome-css) to improve its appearance and readability. The documentation is dynamically updated and deployed on [GitHub Pages](https://p4lang.github.io/p4c/) for easy access and sharing.

Reviewer 2: There appears to be some kind of missing text around `Doxygen.he`

> In P4C, documentation is generated using Doxygen.The generated documentation is enhanced with [Doxygen Awesome CSS](https://github.com/jothepro/doxygen-awesome-css) to improve its appearance and readability. The documentation is dynamically updated and deployed on [GitHub Pages](https://p4lang.github.io/p4c/) for easy access and sharing.

Reviewer: Less marketing speak here :) 

```suggestion
In P4C, documentation is generated using Doxygen.The generated documentation depends on [Doxygen Awesome CSS](https://github.com/jothepro/doxygen-awesome-css). The documentation is dynamically updated and deployed on [GitHub Pages](https://p4lang.github.io/p4c/).
```
